server/parsec/components/memory/account.py

```diff
@@ -45,6 +45,22 @@ def __init__(
         self._data = data
         self._event_bus = event_bus
 
+    @override
+    async def get_password_algorithm_or_fake_it(self, email: EmailAddress) -> PasswordAlgorithm:
+        try:
+            account = self._data.accounts[email]
+            active_authentication_methods = account.current_vault.active_authentication_methods
+        except KeyError:
+            active_authentication_methods = ()
+
+        for auth_method in active_authentication_methods:
+            if auth_method.password_algorithm is not None:
+                return auth_method.password_algorithm
+
+        else:
+            # No password found
+            return self._generate_fake_password_algorithm(email)
```

Comment:
The `else` seems unnecessary since the `for` has no `break` (only a `return`). 
```suggestion
        # Account does not exists or it does not have an active password algorithm.
        # In either case, a fake configuration is returned. 
        return self._generate_fake_password_algorithm(email)
```